ksiren/src/main/java/com/brightspace/ksiren/Action.kt

```diff
@@ -15,14 +15,35 @@ package com.brightspace.ksiren
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+
+interface ActionBase : JsonSerializable {
+	val name: String
+	val classes: List<String>
+	val method: String
+	val href: String
+	val title: String?
+	val type: ContentType
+	val fields: List<FieldBase>
+
+	fun hasField(name: String): Boolean = this.fields.any() { it.name == name }
+
+	fun toJsonRequestBody(): String? =
+		if (fields.isNotEmpty())
+			fields.mapNotNull { field -> field.value?.let { json -> field.name to json } }
+				.joinToString(prefix = "{", postfix = "}") { "\"${it.first}\": \"${JsonUtils.escapeJson(it.second)}\"" }
+		else null
```

Comment:
I would make this a normal function instead of a one-liner (using the `=` syntax) due to the length.